InnerEye/ML/dataset/full_image_dataset.py

```diff
@@ -281,12 +286,15 @@ def convert_channels_to_file_paths(channels: List[str],
 
     for channel_id in channels:
         row = rows.loc[rows[CSV_CHANNEL_HEADER] == channel_id]
-        if len(row) == 0:
+        if len(row) == 0 and not allow_incomplete_labels:
             failed_channel_info += f"Patient {patient_id} does not have channel '{channel_id}'" + os.linesep
+        elif len(row) == 0 and allow_incomplete_labels:
+            # Keeps track of missing channels order
+            paths.append(Path(''))
```

Comment:
I feel that this is not a great choice of return value. If you return None for missing files, you can change the type annotation to Optional[Path], and already by that indicate clearly to the caller that something may be missing. Also, if you work with Path(""), the caller needs to be told clearly about that special value